Tenant search API documentation (#1917)
FusionAuth/fusionauth-site: site/docs/v1/tech/apis/tenants.adoc

* <<Create a Tenant>>
* <<Retrieve a Tenant>>

> * <<Search for Tenants>>

Reviewer: Should we standardize (at least on these new APIs with search) on the order?

https://github.com/FusionAuth/fusionauth-site/pull/1911/files#diff-4d0544aa40a149bba5144812d2f2bd30e279c675c0c0bf5292544b1e6322a7fcR22 is different (has search at the end).

I like CRSUD as the order, but don't really care as long as it is consistent.

Can @mmanes and @spwitt get together and make sure they are consistent?

Author: I reviewed my other PRs and made sure they were consistent.

include::docs/v1/tech/apis/_tenants-response-body.adoc[]

== Search for Tenants

[NOTE.since]
====
This API has been available since 1.44.0
====

This API is used to search for Tenants and may be called using the `GET` or `POST` HTTP methods. Examples of each are provided below. The `POST` method is provided to allow for a richer request object without worrying about exceeding the maximum length of a URL. Calling this API with either the `GET` or `POST` HTTP method will provide the same search results given the same query parameters.

> *Note:* Requests to this API must not include an `X-FusionAuth-TenantId` header, and must be authenticated by an API key which is not scoped to a specific Tenant.

Reviewer: ```suggestion
[NOTE.note]
====
Requests to this API must not include an `X-FusionAuth-TenantId` header, and must be authenticated by a non-tenant scoped API key.
====
```

Reviewer: or else what? Does the request fail? Erroneous results?

Author: It returns a status of 400, with a body of 
```
{
  "fieldErrors": {},
  "generalErrors": [
    {
      "code": "[restricted]",
      "message": "Your request is not allowed. You cannot search for tenants using a tenant restricted request. Omit the X-FusionAuth-TenantId HTTP header if provided, or repeat the request with an API key that is not assigned to a particular tenant."
    }
  ]
}
```
